modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.managers.systemview;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.internal.managers.systemview.walker.ScanQueryViewWalker;
+import org.apache.ignite.internal.processors.cache.GridCacheContext;
+import org.apache.ignite.internal.processors.cache.query.GridCacheQueryManager;
+import org.apache.ignite.internal.util.future.GridFutureAdapter;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.apache.ignite.spi.IgniteSpiCloseableIterator;
+import org.apache.ignite.spi.systemview.view.ScanQueryView;
+import org.apache.ignite.spi.systemview.view.SystemView;
+import org.jetbrains.annotations.NotNull;
+
+import static org.apache.ignite.internal.processors.cache.query.GridCacheQueryType.SCAN;
+import static org.apache.ignite.internal.processors.metric.impl.MetricUtils.metricName;
+
+/**
+ * {@link SystemView} implementation providing data about cache queries.
+ *
+ * @see GridCacheQueryManager#queryIterators()
+ * @see GridCacheQueryManager.ScanQueryIterator
+ * @see ScanQueryView
+ */
+public class ScanQuerySystemView<K, V> extends AbstractSystemView<ScanQueryView> {
+    /** Scan query system view. */
+    public static final String SCAN_QRY_SYS_VIEW = metricName("scan", "queries");
+
+    /** Scan query system view description. */
+    public static final String SCAN_QRY_SYS_VIEW_DESC = "Scan queries";
+
+    /** Cache data. */
+    private Collection<GridCacheContext<K, V>> cctxs;
+
+    /**
+     * @param cctxs Cache data.
+     */
+    public ScanQuerySystemView(Collection<GridCacheContext<K, V>> cctxs) {
+        super(SCAN_QRY_SYS_VIEW, SCAN_QRY_SYS_VIEW_DESC, ScanQueryView.class, new ScanQueryViewWalker());
+
+        this.cctxs = cctxs;
+    }
+
+    /** {@inheritDoc} */
+    @Override public int size() {
+        int sz = 0;
+
+        Iterator<ScanQueryView> iter = new QueryDataIterator();
+
+        while(iter.hasNext())
+            sz++;
```

Comment:
Variable {{iter}} is not updated insude the loop(do not call next()).

Reply:
and missed space

Reply:
I made this intentionally, because, with the next call we generate unnecessary object.
So we can just iterate through data with `hasNext` calls.
It's not pure but reduce GC pressure.

What do you think?

Reply:
In general, I’m for.
I suggest to add javadoc for clarity that every call the `hasNext()` method will switch iterator onto next scan query.

Reply:
JavaDoc added.